core/src/main/kotlin/fr/sncf/osrd/stdcm/graph/STDCMPathfinding.kt

```diff
@@ -156,6 +156,10 @@ class STDCMPathfinding(
             if (Duration.between(start, Instant.now()).toSeconds() >= pathfindingTimeout)
                 throw OSRDError(ErrorType.PathfindingTimeoutError)
             val edge = queue.poll() ?: return null
+            if (edge.weight!!.isInfinite()) {
```

Comment:
Will be changed soon, once I finish transforming code to use nodes rather than edges.

Reply:
Yeah and I assume this will cause a conflict with your branch, you can ignore that version and redo it in a better way.

It was just frustrating to run tests and benchmarks, to see the log printing `heuristic initialized, best theoretical time = +inf`, and then to look for a path until timeout anyway